integration/test/Test/Teams.hs

```diff
@@ -464,3 +465,12 @@ testUpgradeGuestToTeamShouldFail = do
 
   upgradePersonalToTeam guest "wonderland" `bindResponse` \resp -> do
     resp.status `shouldMatchInt` 404
+
+testDeleteTeamUserRatelimitingIsPropagated :: (HasCallStack) => App ()
+testDeleteTeamUserRatelimitingIsPropagated = do
+  (owner, tid, mems) <- createTeam OwnDomain 10
+  -- this is eventually going to run into rate limiting of internal request `/i/users/:uid/reauthenticate`
+  statusCodes <- for mems $ \m -> do
+    bindResponse (deleteTeamMember tid owner m) $ \resp -> do
+      pure resp.status
+  Set.fromList statusCodes `shouldMatchSet` ([202, 429] :: [Int])
```

Comment:
Should we make sure that 429 was returned at least once?

Reply:
This actually asserts that it's returned at least once. But I might not 100% understand your comment?